numpyro/primitives.py

```diff
@@ -316,7 +317,8 @@ def process_message(self, msg):
         cond_indep_stack = msg['cond_indep_stack']
         frame = CondIndepStackFrame(self.name, self.dim, self.subsample_size)
         cond_indep_stack.append(frame)
-        if msg['type'] == 'sample':
+        # only expand if fn is Distribution, not a Funsor
+        if msg['type'] == 'sample' and isinstance(msg['fn'], Distribution):
```

Comment:
Not sure if we have a workaround for this. If `fn` is a Funsor, we can't use the below code to expand it.